src/view/readonlyFileSystemProvider.ts

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+export class ReadonlyFileSystemProvider implements vscode.FileSystemProvider {
+	protected _onDidChangeFile = new vscode.EventEmitter<vscode.FileChangeEvent[]>();
+	onDidChangeFile = this._onDidChangeFile.event;
+
+	constructor() {}
+
+	watch(_uri: vscode.Uri, _options: { recursive: boolean; excludes: string[]; }): vscode.Disposable {
+		/** no op */
+		return { dispose: () => {} };
+	}
+
+	stat(_uri: any): vscode.FileStat {
+		/** mock stat as they are not necessarily needed */
+		return {
+			type: vscode.FileType.File,
+			ctime: 0,
+			mtime: 0,
+			size: 0
+		};
+	}
+
+	readDirectory(_uri: vscode.Uri): [string, vscode.FileType][] {
+		return [];
+	}
+
+	createDirectory(_uri: vscode.Uri): void {
+		/** no op */
+	}
+
+	async readFile(_uri: vscode.Uri): Promise<Uint8Array> {

```

Comment:
Might be nice to make this class abstract and then make `readFile` abstract, since currently the `ReadonlyFileSystemProvider` isn't really usable on its own.